### PR TITLE
fixes #3664 - support deletion of installation media that's in use

### PR DESCRIPTION
**Pull request by isratrade**

### CODE REVIEW

**Comment by lzap** [comment]
LGTM testing.

**Comment by lzap** [comment]
WGFM

**Comment by lzap** [comment]
Once unit tests shows green, this is ready to be merged.

Nitpick: The error message shows "mediaXYZ is used by hostXYZ". The best experience would be to inform the user about that build mode. Less important.

File: app/models/medium.rb [hunk 1]
```diff
@@ -4,11 +4,11 @@ class Medium < ActiveRecord::Base
   include ValidateOsFamily
   audited :allow_mass_assignment => true
 
-  before_destroy EnsureNotUsedBy.new(:hosts, :hostgroups)
+  before_destroy :ensure_hosts_not_in_build
```
Comment (isratrade) [inline app/models/medium.rb:7]: re-factored here to create custom message that "media is used by host in build mode"

**Comment by isratrade** [comment]
It's not clear to me which katello tests fail. 
@jlsherrill, @ehelms, @thomasmckay, can you please have a look?

**Comment by domcleal** [comment]
It timed out.  [test]

**Comment by lzap** [comment]
Merged as 08d248f for Foreman 1.6. Thank you @isratrade !

**Pull request closed without merging**